.github/workflows/build-envoy-builder-image-ci.yaml

```diff
@@ -14,41 +14,46 @@ jobs:
     environment: release-base-images
     steps:
     - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@8c0edbc76e98fa90f69d9a2c020dcb50019dc325
-      with:
-        version: v0.9.1
+      uses: docker/setup-buildx-action@f03ac48505955848960e80bbb68046aa35c7b9e7 # v2.4.1
+
     - name: Cache Docker layers
-      uses: actions/cache@58c146cc91c5b9e778e71775dfe9bf1442ad9a12
+      uses: actions/cache@58c146cc91c5b9e778e71775dfe9bf1442ad9a12 # v3.2.3
       with:
         path: /tmp/buildx-cache
         key: docker-cache-${{ github.head_ref }}
         restore-keys: docker-cache-builder-master
+
     - name: Login to quay.io
-      uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
+      uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a # v2.1.0
       with:
         registry: quay.io
-        username: ${{ secrets.QUAY_ENVOY_USERNAME }}
-        password: ${{ secrets.QUAY_ENVOY_PASSWORD }}
+        username: ${{ secrets.QUAY_ENVOY_USERNAME_DEV }}
+        password: ${{ secrets.QUAY_ENVOY_PASSWORD_DEV }}
+
     - name: Checkout PR
-      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
       with:
         ref: ${{ github.event.pull_request.head.sha }}
+
     - name: Prep for build
       run: |
         echo "${{ github.event.pull_request.head.sha }}" >SOURCE_VERSION
         echo "ENVOY_VERSION=$(cat ENVOY_VERSION)" >> $GITHUB_ENV
         echo "BAZEL_VERSION=$(cat .bazelversion)" >> $GITHUB_ENV
+
     - name: PR Multi-arch build & push of cilium-envoy-builder
-      uses: docker/build-push-action@37abcedcc1da61a57767b7588cb9d03eb57e28b3
+      uses: docker/build-push-action@3b5e8027fcad23fda98b2e3ac259d8d67585f671 # v4.0.0
       id: docker_build_builder_ci
       with:
+        provenance: false
         context: .
         file: ./Dockerfile.builder
         platforms: linux/amd64,linux/arm64
         push: true
-        tags: quay.io/${{ github.repository_owner }}/cilium-envoy-builder:${{ env.BAZEL_VERSION }}
+        tags: quay.io/${{ github.repository_owner }}/cilium-envoy-builder-dev:${{ env.BAZEL_VERSION }}
+
     - name: CI Image Digest
       shell: bash
       run: |
         echo "Digests:"
-        echo "quay.io/${{ github.repository_owner }}/cilium-envoy-builder:${{ env.BAZEL_VERSION }}@${{ steps.docker_build_builder_ci.outputs.digest }}"
+        echo "quay.io/${{ github.repository_owner }}/cilium-envoy-builder-dev:${{ env.BAZEL_VERSION }}@${{ steps.docker_build_builder_ci.outputs.digest }}"
```

Comment:
if we are changing to dev, should we have another one job to push to quay.io/cilium-envoy-builder repository? My original idea is to follow the same way with cilium/cilium-builder, which:

- PR build will require approval, and push to cilium/cilium-builder.
- Author can use the image digest in step 1 in the separate commit in the same PR.

https://github.com/cilium/proxy/blob/d92715404738ac985a7893553e71e0d7203376d0/Dockerfile#L4(https://github.com/cilium/proxy/blob/d92715404738ac985a7893553e71e0d7203376d0/Dockerfile#L4)